src/ghost-content/index.js

```diff
@@ -27,21 +27,23 @@ module.exports = ({ logger, db, projectLocation, enabled }) => {
   const pendingRevisionsByFolder = {}
   const trackedFolders = []
 
-  const upsert = ({ knex, tableName, where, data, idField = 'id' }) =>
-    knex(tableName)
+  const upsert = ({ knex, tableName, where, data, idField = 'id', trx = null }) => {
+    const prepareQuery = () => (trx ? knex(tableName).transacting(trx) : knex(tableName))
+    return prepareQuery()
       .where(where)
       .select(idField)
       .then(res => {
         const id = get(res, '0.id')
         return id
-          ? knex(tableName)
+          ? prepareQuery()
               .where(idField, id)
               .update(data)
-              .then()
-          : knex(tableName)
-              .insert(Object.assign({}, where, data))
-              .then()
+              .then(() => Promise.resolve(id))
```

Comment:
you shouldn't need `Promise.resolve` here, simply return `id`. can also use Bluebird's `thenReturn`
similarly below you don't need resolve

Reply:
Just refactored that thing.